src/components/editTestCase/qdm/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx

```diff
@@ -0,0 +1,310 @@
+  getGenderDataElement,
+  getBirthDateElement,
+  getRaceDataElement,
+} from "./DemographicsSectionConst";
+import { MenuItem as MuiMenuItem } from "@mui/material";
+
+export interface CodeSystem {
+  code: string;
+  display: string;
+  version: string;
+  system: string;
+}
+
+const DemographicsSection = ({ canEdit }) => {
+  dayjs.extend(utc);
+  dayjs.utc().format(); // utc format
+  const formik: any = useFormikContext();
+
+  // select stuff
+  const [qdmPatient, setQdmPatient] = useState<QDMPatient>();
+  // this will be local
+  const [raceDataElement, setRaceDataElement] = useState<DataElement>();
+  const [genderDataElement, setGenderDataElement] = useState<DataElement>();
+  const [birthDateElement, setBirthDateElement] = useState<DataElement>();
+
+  const selectOptions = (options) => {
+    return [
+      options
+        .sort((a, b) => a.display.localeCompare(b.display))
+        .map((opt, i) => {
+          const sanitizedString = opt.display.replace(/"/g, "");
+          return (
+            <MuiMenuItem
+              key={`${sanitizedString}-${i}`}
+              value={sanitizedString}
+            >
+              {sanitizedString}
+            </MuiMenuItem>
+          );
+        }),
+    ];
+  };
+
+  // this populates the json making it able to be edited. we should only do this before change
+  useEffect(() => {
+    let patient: QDMPatient = null;
+    // if (props.currentTestCase?.json) {
+    if (formik.values?.json) {
+      patient = JSON.parse(formik.values.json);
+    }
+    // save local patient
+    if (patient) {
+      setQdmPatient(patient);
+      const dataElements: DataElement[] = patient.dataElements;
+
+      // what is the purpose of this intermediate
+      dataElements.forEach((element) => {
+        if (element.qdmStatus === "race") {
+          setRaceDataElement(element);
+        }
+        if (element.qdmStatus === "gender") {
+          setGenderDataElement(element);
+        }
+        // seems unecessary?
+        if (element.qdmStatus === "birthdate") {
+          setBirthDateElement(element);
+        }
+      });
+    } else {
+      // default values for QDM patient. to do race and gender default values?
+      const newRaceDataElement: DataElement = getRaceDataElement(
+        "American Indian or Alaska Native"
+      );
+      setRaceDataElement(newRaceDataElement);
+      const newGenderDataElement: DataElement = getGenderDataElement("Female");
+      setGenderDataElement(newGenderDataElement);
+      // unneeded ?
+      const newbirthDateElement = getBirthDateElement(dayjs());
+      setBirthDateElement(newbirthDateElement);
+
+      const patient: QDMPatient = new QDMPatient();
+      patient.qdmVersion = "5.6";
+      setQdmPatient(patient);
+    }
+  }, [formik.values.json]);
+
+  // given element and type make this return instead
+  const generateNewQdmPatient = (newElement: DataElement, type: string) => {
+    const patient: QDMPatient = new QDMPatient();
```

Comment:
we create a new QDMPatient whenever something changes. we'll lose the id of the original patient. we could use a spread operator instead, or are we planning to regenerate id later?